### DISCUSSION TITLE
Could we support session storage as well?

### DISCUSSION THREAD

**Discussion by kayac-chang** [aklinker1/webext-core#45] [Ideas]
First, I want to thank @aklinker1 so much for the work!
I joined to develop the browser extension recently 
but found tons of packages that need to be updated.
This package shines because I finally found a solution to do the unit testing.

But I figured out the package `@webext-core/fake-browser` didn't support the [session storage](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/session) that my project was based on.

Could we also support session storage as well?
I can also help to contribute this feature if the team wants.

**Comment by aklinker1** [comment]
Oh wow, looks like Firefox recently added support for it!

I'll add it to the library, but in the meantime, you can do it yourself with `defineExtensionStorage`:

```ts
export const sessionExtStorage = defineExtensionStorage(browser.storage.session)
```

`webextension-polyfill` might complain if you're using typescript, but it should exist at runtime, and you can ignore that error.

**Reply by kayac-chang** [reply]
Sure, thank you @aklinker1.
Currently, I fork the repo and open a branch to support this feature.
It works fine now.

**Reply by aklinker1** [reply]
Open a PR and I can merge and release it to NPM.

**Reply by kayac-chang** [reply]
Sure, I just opened it https://github.com/aklinker1/webext-core/pull/46.

**Reply by aklinker1** [reply]
I added `sessionExtStorage` to `@webext-core/storage` [`v1.2.0`](https://github.com/aklinker1/webext-core/releases/tag/storage-v1.2.0), and I released your PR as apart of `@webext-core/fake-browser` [`v1.3.0`](https://github.com/aklinker1/webext-core/releases/tag/fake-browser-v1.3.0)
